lsp/src/cli.ml

```diff
@@ -11,25 +11,39 @@ module Arg = struct
     ; mutable port : int option
     ; mutable stdio : bool
     ; mutable spec : (string * Arg.spec * string) list
+    ; mutable clientProcessId : int option
     }
 
   let create () =
-    let t = { pipe = None; port = None; stdio = false; spec = [] } in
+    let t =
+      { pipe = None
+      ; port = None
+      ; stdio = false
+      ; spec = []
+      ; clientProcessId = None
+      }
+    in
     let spec =
       [ ("--pipe", Arg.String (fun p -> t.pipe <- Some p), "set pipe path")
       ; ("--socket", Arg.Int (fun p -> t.port <- Some p), "set port")
       ; ("--stdio", Arg.Unit (fun () -> t.stdio <- true), "set stdio")
       ; ( "--node-ipc"
         , Arg.Unit (fun () -> raise @@ Arg.Bad "node-ipc isn't supported")
         , "not supported" )
+      ; ( "--clientProcessId"
+        , Arg.Int (fun pid -> t.clientProcessId <- Some pid)
+        , "set the pid of the lsp client" )
```

Comment:
I wonder if we should add an explicit remark about client pid being ignored. Currently, it's only mentioned in PR description. Maybe we should add it here?

Reply:
Note that this code is part of our general purpose lsp library. So we can't make assumptions whether it's being used or not.